### ISSUE TITLE
Keyboard events are lost when notification is dismissed

### ISSUE THREAD

**Issue by fantasticsid** [electron/electron#9190]
* Electron version: 1.4.12
* Operating system: Windows 10

### Expected behavior
Typing into input box should not be affected by notifications being dismissed

### Actual behavior

While I'm typing in an input box, if a notification is dismissed, I lose 1 or 2 characters. I shouldn't lose characters being typed in.

### How to reproduce
This is fully reproducible :)

- get a powershell open and ready
- open Discord (built with electron). Open the console, then focus on any input box within Discord app

in Discord's console, type in:

```
title="some title"
body = {body: 'notification text', tag: 123, dir: 'auto', lang: 'en', silent: true}
new Notification(title, body)
```
you'll see a notification pops up

now quickly go to powershell and run this script: (save it in a script so it's faster)

```
add-type -AssemblyName microsoft.VisualBasic
add-type -AssemblyName System.Windows.Forms

[Microsoft.VisualBasic.Interaction]::AppActivate("Discord")

For ($i=0; $i -le 100; $i++) {
  [System.Windows.Forms.SendKeys]::SendWait("V")
  start-sleep -Milliseconds 30
  [System.Windows.Forms.SendKeys]::SendWait("O")
  start-sleep -Milliseconds 30
  [System.Windows.Forms.SendKeys]::SendWait("I")
  start-sleep -Milliseconds 30
  [System.Windows.Forms.SendKeys]::SendWait("D")
  start-sleep -Milliseconds 30
  [System.Windows.Forms.SendKeys]::SendWait(" ")
  start-sleep -Milliseconds 30
}
```
(the script just focuses discord and try to send keyboard events to it, simulating user typing in the app)

Right when the notification is dismissed, you'll likely see 1 or 2 characters missing. Please try it out and be patient about reproducing this, if it does not reproduce, try restarting app.

Also it's not just Discord, it's also fully reproducible in my app.

![whatsapp image 2017-04-13 at 5 44 30 pm](https://cloud.githubusercontent.com/assets/203771/25029455/4f456e54-2072-11e7-81e6-f5dba3fa0c92.jpeg)

**Comment by fantasticsid** [comment]
I recorded a video reproducing this. Still using the same power shell script to reproduce it (it's also reproducible manually). 

https://youtu.be/T9id4GAYhrA

**Comment by fantasticsid** [comment]
To provide more info: in vendor/brightray/browser/win/windows_toast_notification.cc, if I don't register handler for 'dismissed' event by commenting out the call to
```
toast->add_Dismissed
```
it seems to fix the problem.

Obviously this is not a fix, just more evidence that dismiss handler might be the problem.

**Comment by kevinsawicki** [comment]
> Operating system: Windows


What version(s) of Windows are you seeing this on? Looks like Windows 10 from your video.

**Comment by fantasticsid** [comment]
yeah it's on Windows 10

**Comment by zcbenz** [comment]
A few questions:
1. Does the same problem happen if you open a simple page in Electron like below?

```html
<body>
<input/>
<script>document.getElementsByTagName('input')[0].focus()</script>
</body>
```

2. Does the same problem happen for Chrome browser?

**Comment by fantasticsid** [comment]
I tried again today and couldn't reproduce, either in Discord or from a simple page in vanilla Electron. It's not always reproducible, but when it happened it happened consistently.

I don't think it happens in Chrome. Chrome has a different notification implementation though.

Could the event loop integration somehow interact weirdly with Notification callback?

Also it seemed to happen under heavier load (high cpu/memory usage) which suggests a race

**Comment by sofianguy** [comment]
Thanks for contributing!

We are no longer releasing bug fixes for versions of Electron older than 1.6x. We prefer to only keep issues that are on our immediate radar.

If you are still having the same issue in an up-to-date version, feel free to reach out and we'll reopen.
